crates/signature-mpc/src/twopc_mpc_protocols/dwallet_transfer.rs

```diff
@@ -0,0 +1,361 @@
+    let public_key = bcs::to_bytes(&public_key).unwrap();
+    (public_key, private_key)
+}
+
+pub fn get_proof_public_parameters(pub_key: Vec<u8>) -> LangPublicParams {
+    let secp256k1_scalar_public_parameters = secp256k1::scalar::PublicParameters::default();
+
+    let secp256k1_group_public_parameters = secp256k1::group_element::PublicParameters::default();
+
+    let paillier_public_parameters: tiresias::encryption_key::PublicParameters =
+        bcs::from_bytes(&pub_key).unwrap();
+
+    let generator = secp256k1_group_public_parameters.generator;
+
+    encryption_of_discrete_log::PublicParameters::<
+        { tiresias::PLAINTEXT_SPACE_SCALAR_LIMBS },
+        { secp256k1::SCALAR_LIMBS },
+        secp256k1::GroupElement,
+        EncryptionKey,
+    >::new::<
+        { tiresias::PLAINTEXT_SPACE_SCALAR_LIMBS },
+        { secp256k1::SCALAR_LIMBS },
+        secp256k1::GroupElement,
+        EncryptionKey,
+    >(
+        secp256k1_scalar_public_parameters,
+        secp256k1_group_public_parameters,
+        paillier_public_parameters,
+        generator,
+    )
+}
+
+fn parse_plaintext(
+    plaintext: Vec<u8>,
+    public_parameters: &tiresias::encryption_key::PublicParameters,
+) -> PlaintextSpaceGroupElement {
+    let plaintext = pad_vector(plaintext);
+    let plaintext: LargeBiPrimeSizedNumber = LargeBiPrimeSizedNumber::from_be_slice(&plaintext);
+    let plaintext = PlaintextSpaceGroupElement::new(
+        plaintext,
+        public_parameters.plaintext_space_public_parameters(),
+    )
+    .unwrap();
+    plaintext
+}
+
+pub fn generate_proof(
+    encryption_key: Vec<u8>,
+    plaintext: Vec<u8>,
+    language_public_parameters: encryption_of_discrete_log::PublicParameters<
+        PLAINTEXT_SPACE_SCALAR_LIMBS,
+        SCALAR_LIMBS,
+        twopc_mpc::secp256k1::GroupElement,
+        EncryptionKey,
+    >,
+) -> ProofPublicOutput {
+    let paillier_public_parameters: tiresias::encryption_key::PublicParameters =
+        bcs::from_bytes(&encryption_key).unwrap();
+
+    let unbounded_witness_public_parameters = language_public_parameters
+        .randomness_space_public_parameters()
+        .clone();
+
+    let plaintext = parse_plaintext(plaintext, &paillier_public_parameters);
+
+    let randomness = RandomnessSpaceGroupElement::sample(
+        language_public_parameters
+            .encryption_scheme_public_parameters
+            .randomness_space_public_parameters(),
+        &mut OsRng,
+    )
+    .unwrap();
+
+    let enhanced_language_public_parameters = enhanced_language_public_parameters::<
+        { maurer::SOUND_PROOFS_REPETITIONS },
+        RANGE_CLAIMS_PER_SCALAR,
+        RandomnessSpaceGroupElement,
+        Lang,
+    >(
+        unbounded_witness_public_parameters,
+        language_public_parameters,
+    );
+
+    let witnesses = generate_witnesses(randomness, &enhanced_language_public_parameters, plaintext);
+
+    let (proofs, statements) = SecretShareProof::prove(
+        &PhantomData,
```

Comment:
This should be `ProtocolContext`. Need to think what to put there

Reply:
following a conversation with our cryptographers, it seems fine to leave it as-is for this purpose